### PR TITLE
New landingpage

### PR DESCRIPTION
**Pull request by johan-lindell**

### CODE REVIEW

**Comment by johan-lindell** [comment]
för dehär behöver man ännu adda en credentials.json service account fil till rooten för att googleapi skall fungera, skulle vara bra att skapa en sån med en av tfs egan google accounts

File: pages/api/calendar.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import fs from 'fs/promises'
+import path from 'path'
+import { calendar_v3, google } from 'googleapis'
+import { CalendarEvent } from '../../models/event'
+import { NextApiRequest, NextApiResponse } from 'next/types'
+
+// If modifying these scopes, update the service account permissions.
+const SCOPES = ['https://www.googleapis.com/auth/calendar.readonly']
```
Comment (isofore) [inline pages/api/calendar.ts:8]:
```suggestion
const SCOPES = ['https://www.googleapis.com/auth/calendar.readonly'] as const
```

File: pages/api/calendar.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ * Loads the service account credentials from the JSON file.
+ *
+ * @return {Promise<Object>}
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function loadServiceAccountCredentials(): Promise<any> {
+  const content = await fs.readFile(CREDENTIALS_PATH)
+
+  return JSON.parse(content.toString())
+}
```
Comment (isofore) [inline pages/api/calendar.ts:20]: If I remember correctly the Google module should be able to load the credentials implicitly with an environmental variable.

File: pages/api/calendar.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+): Promise<calendar_v3.Schema$Event[]> {
+  const calendar = google.calendar({ version: 'v3', auth })
+
+  const parsedDate = Date.parse(date)
+
+  const res = await calendar.events.list({
+    calendarId,
+    //load dates 2 months back and 3 forward
+    timeMin: new Date(parsedDate - 1 * 30 * 24 * 60 * 60 * 1000).toISOString(),
+    timeMax: new Date(parsedDate + 3 * 30 * 24 * 60 * 60 * 1000).toISOString(),
```
Comment (isofore) [inline pages/api/calendar.ts:41]: 2 months back but you have 1 in the code is this correct?

File: pages/api/calendar.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+): Promise<CalendarEvent[]> => {
+  try {
+    const credentials = await loadServiceAccountCredentials()
+    const auth = new google.auth.JWT(
+      credentials.client_email,
+      undefined,
+      credentials.private_key,
+      SCOPES
+    )
+    await auth.authorize()
```
Comment (isofore) [inline pages/api/calendar.ts:67]: Google auth should happen implicitly

File: pages/events/[slug].tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+const EventPage: NextPage<Props> = ({
+  event,
+  isHomePage,
+  logos,
+  navbarLinks,
+}) => (
+  <>
+    <header>
+      <Header navbarLinks={navbarLinks} isHomePage={isHomePage} />
+    </header>
```
Comment (isofore) [inline pages/events/[slug].tsx:38]: I would move the `header` tags inside of `Header`

File: pages/index.tsx [hunk 6]
```diff
@@ -23,37 +28,55 @@ type Props = {
...
+              width={1000}
+              style={{ objectFit: 'cover' }}
+            />
+            <Image
+              src={`/images/banner/1.jpg`}
+              alt="jeej"
+              height={500}
+              width={1000}
+              style={{ objectFit: 'cover' }}
+            />
```
Comment (isofore) [inline pages/index.tsx:71]: These images should be loaded from Strapi

**Review by isofore — COMMENTED** [review]
Great work! I would recheck formatting (`./node_modules/.bin/prettier -c components lib models pages utils`). There are also quite a lot of places where you could add an extra newline and capitalize some components.

**Review by isofore — COMMENTED** [review]
Great work! I would recheck formatting (`./node_modules/.bin/prettier -c components lib models pages utils`). There are also quite a lot of places where you could add an extra newline and capitalize some components.

File: pages/api/calendar.ts (hunk 2 above)
Comment (johan-lindell) [reply pages/api/calendar.ts:20]: Google does this on their own quickstart guide: https://developers.google.com/calendar/api/quickstart/nodejs

Comment (backjonas) [reply pages/api/calendar.ts:20]: Could probably make deployment simpler if all configurable vars were in one env file, something like this seems to work
https://cloud.google.com/nodejs/docs/reference/google-auth-library/latest#loading-credentials-from-environment-variables

**Pull request merged**